filip/clients/ngsi_v2/iota.py

```diff
@@ -163,19 +164,15 @@ def get_group(self, *, resource: str, apikey: str) -> ServiceGroup:
         Returns:
 
         """
-        url = urljoin(self.base_url, 'iot/services')
-        headers = self.headers
-        params = {'resource': resource,
-                  'apikey': apikey}
-
-        try:
-            res = self.get(url=url, headers=headers, params=params)
-            if res.ok:
-                return ServiceGroup(**res.json()['services'][0])
-            res.raise_for_status()
-        except requests.RequestException as err:
-            self.log_error(err=err, msg=None)
-            raise
+        groups = self.get_group_list()
+        groups = filter_group_list(group_list=groups, resources=resource, apikeys=apikey)
+        if len(groups) == 1:
+            group = groups[0]
+            return group
+        elif len(groups) == 0:
+            raise KeyError(f"Service group with resource={resource} and apikey={apikey} was not found")
```

Comment:
I am not sure with the KeyError here. Maybe there is a better Error type or even another way to inform the users.

Reply:
Well, you can use a ValueError, but in the very end you are looking for a specific combination of keys ;)

Reply:
Thats true. Would it be better, if we build a custom error type here? Something like "GroupNotFoundError".

Reply:
@djs0109 I suggest to stick to the keyError

Reply:
OK, then I will resolve this point ;)